openfeature/api.py

```diff
@@ -14,31 +14,52 @@
 
 _hooks: typing.List[Hook] = []
 
+_providers: typing.Dict[str, FeatureProvider] = {}
+
 
 def get_client(
-    name: typing.Optional[str] = None, version: typing.Optional[str] = None
+    domain: typing.Optional[str] = None, version: typing.Optional[str] = None
 ) -> OpenFeatureClient:
-    return OpenFeatureClient(name=name, version=version, provider=_provider)
+    return OpenFeatureClient(domain=domain, version=version)
 
 
-def set_provider(provider: FeatureProvider) -> None:
-    global _provider
+def set_provider(
+    provider: FeatureProvider, domain: typing.Optional[str] = None
+) -> None:
     if provider is None:
         raise GeneralError(error_message="No provider")
+
+    if domain:
+        _set_domain_provider(domain, provider)
+        return
+
+    global _provider
     if _provider:
         _provider.shutdown()
     _provider = provider
     provider.initialize(_evaluation_context)
 
 
-def get_provider() -> typing.Optional[FeatureProvider]:
-    global _provider
-    return _provider
+def _set_domain_provider(domain: str, provider: FeatureProvider) -> None:
+    if domain in _providers:
+        old_provider = _providers[domain]
+        del _providers[domain]
```

Comment:
if you want to use Python 3.8 syntax
```suggestion
    if old_provider := _providers.pop(domain, None):
```

Reply:
I like this, but the linter forced me to put the nested if in the same line and it was not as readable as the current version.